OpenBLAS on LoongArch64: openblas_config.h: error: '_Float16' is not supported on this target

I encountered the same error as in #5700 when building FLINT on LoongArch64:

```
configure:15030: checking for cblas.h
configure:15030: gcc -c   -I/nix/store/s3qryikb131hghk6qnakyhrvhwn2v5vp-gmp-with-cxx-6.3.0/include -I/nix/store/29b1k4wdvn1ig9rwnrhjvgp8k5n4njw5-mpfr-4.2.2/include -I/nix/store/lmzx27k35xrg1rdpwc3m>
In file included from /nix/store/h52mz21p1k513nyfvxv40gsllqw29dqi-openblas-0.3.31-dev/include/cblas.h:33,
                 from conftest.c:58:
/nix/store/h52mz21p1k513nyfvxv40gsllqw29dqi-openblas-0.3.31-dev/include/openblas_config.h:63:9: error: '_Float16' is not supported on this target
   63 | typedef _Float16 hfloat16;
      |         ^~~~~~~~
configure:15030: $? = 1
```

This issue didn't happen before https://github.com/NixOS/nixpkgs/pull/450106 where OpenBLAS was updated to 0.3.31.
@martin-frbg Could you help me, please?

--- a/openblas_config_template.h
+++ b/openblas_config_template.h
@@ -40,7 +40,7 @@ typedef uint16_t bfloat16;
 #endif
 
 #if defined(__GNUC__) && (__GNUC__ > 12)
-#if defined(OPENBLAS_ARCH_POWER)
+#if defined(OPENBLAS_ARCH_POWER) || defined(OPENBLAS_ARCH_LOONGARCH64)
 typedef bfloat16 hfloat16;
 #else
 typedef _Float16 hfloat16;

